pkg/registry/apps/statefulset/strategy.go

```diff
@@ -93,6 +97,17 @@ func (statefulSetStrategy) PrepareForUpdate(ctx context.Context, obj, old runtim
 
 }
 
+func dropDisabledFields(newSet, oldSet *apps.StatefulSet) {
+	// The new spec must always be non-nil.
+	if newSet == nil {
+		newSet = &apps.StatefulSet{}
+	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.StatefulSetAutoDeletePVC) && (oldSet == nil || oldSet.Spec.PersistentVolumeClaimDeletePolicy == apps.Retain) {
+		newSet.Spec.PersistentVolumeClaimDeletePolicy = apps.Retain
```

Comment:
I think this can be addressed in the API pr, but policy should be a pointer, and then dropping it would be setting it to nil.